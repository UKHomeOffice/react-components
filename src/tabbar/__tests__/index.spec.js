import React from 'react'
import {render} from 'enzyme'
import {TabBar} from 'govuk-react-components/tabbar';
import {TabItem} from 'govuk-react-components/tabbar/tabitem';

describe('The TabBar Component', () => {
  it('should render children that are passed in', () => {
    expect(
      render(<TabBar>Sample Text</TabBar>)
    ).toMatchSnapshot()
  });

  it('should render an included classname', () => {
    expect(
      render(<TabBar className="class name">Sample Text</TabBar>)
    ).toMatchSnapshot()
  })

  it('should render with a vertical class name with a "vertical" prop', () => {
    expect(
      render(<TabBar className="class name" vertical={true}>Sample Text</TabBar>)
    ).toMatchSnapshot()
  })

  describe('With NavItems', () => {
    it('should render a navitem inside a subnavbar', () => {
      expect(
        render(
          <TabBar>
            <TabItem href="#test">Item A</TabItem>
          </TabBar>
        )
      ).toMatchSnapshot()
    });

    it('should render multiple navitems inside a subnavbar', () => {
      expect(
        render(
          <TabBar>
            <TabItem href="#itema">Item A</TabItem>
            <TabItem href="#itemb">Item B</TabItem>
          </TabBar>
        )
      ).toMatchSnapshot();
    })

    it('should render an included classname', () => {
      expect(
        render(
          <TabBar>
            <TabItem href="#test" className="class-name">Item A</TabItem>
          </TabBar>
        )
      ).toMatchSnapshot()
    });

    it('should render with an active class if the active flag is true', () => {
      expect(
        render(
          <TabBar>
            <TabItem href="#test" className="class-name" active={true}>Item A</TabItem>
          </TabBar>
        )
      ).toMatchSnapshot()
    })
  })
});

