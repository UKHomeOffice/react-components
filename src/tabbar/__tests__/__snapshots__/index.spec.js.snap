// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TabBar Component With NavItems should render a navitem inside a subnavbar 1`] = `
<ul
  class="tab__nav horizontal "
>
  <li>
    <a
      class=""
      href="#test"
    >
      Item A
    </a>
  </li>
</ul>
`;

exports[`The TabBar Component With NavItems should render an included classname 1`] = `
<ul
  class="tab__nav horizontal "
>
  <li>
    <a
      class=" class-name"
      href="#test"
    >
      Item A
    </a>
  </li>
</ul>
`;

exports[`The TabBar Component With NavItems should render multiple navitems inside a subnavbar 1`] = `
<ul
  class="tab__nav horizontal "
>
  <li>
    <a
      class=""
      href="#itema"
    >
      Item A
    </a>
  </li>
  <li>
    <a
      class=""
      href="#itemb"
    >
      Item B
    </a>
  </li>
</ul>
`;

exports[`The TabBar Component With NavItems should render with an active class if the active flag is true 1`] = `
<ul
  class="tab__nav horizontal "
>
  <li>
    <a
      class="active class-name"
      href="#test"
    >
      Item A
    </a>
  </li>
</ul>
`;

exports[`The TabBar Component should render an included classname 1`] = `
<ul
  class="tab__nav horizontal  class name"
>
  Sample Text
</ul>
`;

exports[`The TabBar Component should render children that are passed in 1`] = `
<ul
  class="tab__nav horizontal "
>
  Sample Text
</ul>
`;

exports[`The TabBar Component should render with a vertical class name with a "vertical" prop 1`] = `
<ul
  class="tab__nav vertical  class name"
>
  Sample Text
</ul>
`;
